torch/_dynamo/source.py

```diff
@@ -63,6 +63,8 @@ def name(self):
 class LocalInputSource(LocalSource):
     pos: int
 
+    dynamic_spec: Optional[Tuple[Optional[str], ...]]   # e.g. ['batch', 'seq', None]
```

Comment:
Not sure if `LocalInputSource` is the right place to store this info.

Reply:
I think it's the wrong place. There should be some data structure in Dynamo which stores the VariableTrackers for local inputs, that data structure seems like the right place.

Reply:
I see that Source is convenient because we already propagate it through the meta converter. But sorry, it's still the wrong place to put it. Source is a data structure solely for describing where tensors came from, shouldn't hold other stuff.

Reply:
If we don't reuse LocalInputSource, then I have to plumb it through 7 files https://github.com/pytorch/pytorch/pull/93820
which is uhhhh... 

Also, only input tensor could have dynamic_spec, adding dynamic_spec to VariableBuilder and VariableTracker doesn't looks correct. 

Maybe we should extend the meaning of LocalInputSource?